tsweb/jsonhandler.go

```diff
@@ -16,119 +15,49 @@ type response struct {
 	Data   interface{} `json:"data,omitempty"`
 }
 
-func responseSuccess(data interface{}) *response {
-	return &response{
-		Status: "success",
-		Data:   data,
-	}
-}
-
-func responseError(e string) *response {
-	return &response{
-		Status: "error",
-		Error:  e,
-	}
-}
+// JSONHandlerFunc only take *http.Request as argument to avoid any misuse of http.ResponseWriter.
+// The function's results must be (status int, data interface{}, err error)
+type JSONHandlerFunc func(r *http.Request) (status int, data interface{}, err error)
```

Comment:
As you're introducing a type here, you can hang a method off it directly and skip the `JSONHandler` function. That is:

```
func (f JSONHandlerFunc) ServeHTTP(w ResponseWriter, r *Request) {
    // .. put implementation here that calls f().
}
```